src/information_panel.rs

```diff
@@ -0,0 +1,459 @@
+    ///
+    /// # Returns
+    /// The modified `InformationPanel` instance.
+    pub fn add_metadata_loader(
+        mut self,
+        extension: &str,
+        load_metadata: impl FnMut(&mut IndexMap<String, String>, &PathBuf) + 'static,
+    ) -> Self {
+        self.additional_meta_files
+            .insert(extension.to_string(), Box::new(load_metadata));
+        self
+    }
+
+    /// Reads a preview of the file if it is detected as a text file.
+    fn load_text_file_preview(path: PathBuf, max_chars: usize) -> io::Result<String> {
+        let mut file = File::open(path)?;
+        let mut chunk = [0; 96]; // Temporary buffer
+        let mut buffer = String::new();
+
+        // Add the first chunk to the buffer as text
+        let mut total_read = 0;
+
+        // Continue reading if needed
+        while total_read < max_chars {
+            let bytes_read = file.read(&mut chunk)?;
+            if bytes_read == 0 {
+                break; // End of file
+            }
+            let chars_read: String = String::from_utf8(chunk[..bytes_read].to_vec())
+                .map_err(|_| io::Error::from(io::ErrorKind::InvalidData))?;
+            total_read += chars_read.len();
+            buffer.push_str(&chars_read);
+        }
+
+        Ok(buffer.to_string())
+    }
+
+    fn load_content(&self, path: PathBuf) -> Option<String> {
+        if self.load_text_content {
+            Self::load_text_file_preview(path, self.text_content_max_chars).ok()
+        } else {
+            None
+        }
+    }
+
+    /// Renders the Information Panel in the provided UI context.
+    ///
+    /// # Arguments
+    /// - `ui`: The UI context where the panel will be rendered.
+    /// - `file_dialog`: A reference to the current file dialog, which provides the active file entry.
+    pub fn ui(&mut self, ui: &mut Ui, file_dialog: &mut FileDialog) {
+        const SPACING_MULTIPLIER: f32 = 4.0;
+
+        ui.label("Information");
+        ui.separator();
+
+        // Display metadata in a grid format
+        let width = file_dialog.config_mut().right_panel_width.unwrap_or(100.0) / 2.0;
+
+        if let Some(item) = file_dialog.active_entry() {
+            // load file content and additional metadata if it's a new file
+            self.load_meta_data(item);
+
+            // show preview of selected item
+            self.display_preview(ui, item);
+
+            let spacing = ui.ctx().style().spacing.item_spacing.y * SPACING_MULTIPLIER;
+            ui.separator();
+
+            ui.add_space(spacing);
+
+            // show all metadata
+            self.display_meta_data(ui, width, item);
+        }
+    }
+
+    fn display_preview(&mut self, ui: &mut Ui, item: &DirectoryEntry) {
+        if item.is_dir() {
+            // show folder icon
+            ui.vertical_centered(|ui| {
+                ui.label(egui::RichText::from(item.icon()).size(120.0));
```

Comment:
The size should not be hard coded but should be determined by the size available